Userland/Utilities/json.cpp

```diff
@@ -19,8 +22,33 @@
 #include <LibMain/Main.h>
 #include <unistd.h>
 
-static JsonValue query(JsonValue const& value, Vector<StringView>& key_parts, size_t key_index = 0);
+class SingleValueKey : public RefCounted<SingleValueKey> {
+public:
+    SingleValueKey(StringView key)
+        : m_key(key)
+    {
+    }
+
+    static NonnullRefPtr<SingleValueKey> parse(StringView);
+    ErrorOr<JsonValue> select(JsonValue const& value);
+
+private:
+    StringView m_key;
+};
+
+class MultiValueKey : public RefCounted<MultiValueKey> {
+public:
+    static ErrorOr<NonnullRefPtr<MultiValueKey>> parse(StringView const&);
+    virtual ~MultiValueKey() = default;
+    virtual Vector<JsonValue> select(JsonValue const& value) = 0;
```

Comment:
Should this be `ErrorOr<Vector<JsonValue>>` , like it is in `SingleValueKey`? I.e. it will be an empty array when error occurs. Is an empty match an error, it is ambiguous now?